### PR TITLE
Native module support

### PR DESCRIPTION
**Pull request by Frityet**
This PR adds support for native modules, where you can require `.so` or `.wasm` files and it will be required as a lua module

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,7 +5,7 @@
 *.cjs
 *.d.mts
 *.d.cts
-/dist
+/dist/
```
Comment (Doridian) [inline .gitignore:8]: Why this change?

File: src/Makefile [hunk 2]
```diff
@@ -58,6 +58,7 @@ BINARY_ARGS=-s EXPORTED_FUNCTIONS="[\
 '_lua_type'\
 ]" -s EXPORTED_RUNTIME_METHODS=cwrap,stringToNewUTF8,lengthBytesUTF8,stringToUTF8,UTF8ToString \
 -s WASM -s TOTAL_MEMORY=134217728 -s ASYNCIFY -s ALLOW_TABLE_GROWTH \
+-s MAIN_MODULE\
```
Comment (Doridian) [inline src/Makefile:61]: Space before the `\` please

File: .gitignore (hunk 1 above)
Comment (Frityet) [reply .gitignore:8]: Accident, ill revert

File: src/Makefile (hunk 2 above)
Comment (Frityet) [reply src/Makefile:61]: Will do
